var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -52,10 +57,59 @@ class Mesa(AutotoolsPackage):
     depends_on('libxfixes')
 
     depends_on('glproto@1.4.14:', type='build')
-    depends_on('dri2proto@2.6:', type='build')
-    depends_on('dri3proto@1.0:', type='build')
+    depends_on('dri2proto@2.6:', when='+dri', type='build')
+    depends_on('dri3proto@1.0:', when='+dri', type='build')
     depends_on('presentproto@1.0:', type='build')
     depends_on('pkg-config@0.9.0:', type='build')
+    depends_on('llvm', when='+osmesa', type='build')
 
+    # adamjstewart
     # TODO: Add package for systemd, provides libudev
     # Using the system package manager to install systemd didn't work for me
+    # justiceformikebrown
+    # Added:
+    # ~dri will disable dri on older Centos /RHEL6 servers
+    # which lack newer udev
+    # You must also yum install libsysfs-devel
+    # https://www.mesa3d.org/osmesa.html
+    # https://www.mesa3d.org/autoconf.html
+    def configure_args(self):
+        spec = self.spec
+        version = distro.linux_distribution()[1]
+        distname = distro.linux_distribution()[0]
+        strversion = version.encode('utf-8')
+        # centos / rhel 6
+        if distname == "RedHatEnterpriseServer":
+            if strversion.find('6.0:6.9') == -1:
```

Comment:
Ha ha!  @adamjstewart 

No, and actually Red Hat I believe did use that numbering before Enterprise Linux became a thing.